cmake/PCHSupport.cmake

```
@@ -0,0 +1,27 @@
+option (USE_PCH "Use pre-compiled headers when possible to speed up build" YES)
+
+# PCH support is only available from cmake 3.16 onwards
+if(NOT ${CMAKE_VERSION} VERSION_GREATER "3.16" AND USE_PCH)
+  message(STATUS "Pre-compile headers only available from cmake version >= 3.16, turning OFF")
+  set(USE_PCH NO)
+endif()
+
+function(init_pch_support shlib headers)
+  if (NOT USE_PCH)
+    return()
+  endif()
+  target_precompile_headers(${shlib} PRIVATE ${headers})
+  set(_pch_pie_target ${shlib}_pch_pie)
+  set(_pch_pie_target ${_pch_pie_target} PARENT_SCOPE)
+  file(TOUCH ${CMAKE_CURRENT_BINARY_DIR}/empty.cpp)
+  add_library(${_pch_pie_target} STATIC ${CMAKE_CURRENT_BINARY_DIR}/empty.cpp)
+  set_target_properties(${_pch_pie_target} PROPERTIES PROSITION_INDEPENDENT_CODE OFF)
```

```suggestion
  set_target_properties(${_pch_pie_target} PROPERTIES POSITION_INDEPENDENT_CODE OFF)
```
Silly that CMake doesn't warn about this typo.

---

Yeap, that should result in some warning of some type...

Since the target was a static library the property was already OFF by default, but I guess it doesn't hurt to make it explicit anyway, so I kept it with the typo corrected.